app/lib/frontend/search_service.dart

```diff
@@ -24,73 +30,104 @@ void registerSearchService(SearchService s) => ss.register(#_search, s);
 
 /// A wrapper around the Custom Search API, used for searching for pub packages.
 class SearchService {
-  Future<SearchResultPage> search(SearchQuery query) async {
-    final result = await searchClient.search(query);
+  Future<SearchResultPage> search(SearchQuery query,
+      {bool fallbackToNames = true}) async {
+    PackageSearchResult result;
+    try {
+      result = await searchClient.search(query);
+    } catch (e, st) {
+      _logger.severe('Unable to search packages', e, st);
+    }
+    if (result == null && fallbackToNames) {
+      result = await _fallbackSearch(query);
+    }
     return _loadResultForPackages(query, result.totalCount, result.packages);
   }
 
+  Future<PackageSearchResult> _fallbackSearch(SearchQuery query) async {
+    final names =
+        await nameTracker.getPackageNames().timeout(Duration(seconds: 5));
+    List<PackageScore> scores = <PackageScore>[];
+    final text = (query.query ?? '').trim().toLowerCase();
+    if (text.isNotEmpty) {
```

Comment:
Not this time. I think it would make sense if the NameTracker would implement a search interface (e.g. through a wrapper class), which could do that, but at this time I think it is better to have something to happen other than a 500.